### DISCUSSION TITLE
Data Loss Prevention / Disabling Copying to Clipboard

### DISCUSSION THREAD

**Discussion by mklebrasseur** [coder/code-server#6159] [General]
I know you can prevent downloading files, is there a way to disabling copy from within the workspace?

**Comment by code-asher** [comment]
There is no way to disable copy at the moment although I think it 
would be fairly straightforward, might even be able to allow 
copying in a way that allows you to paste copied text inside 
code-server but not outside the browser which would avoid at least 
some of the productivity hit disabling copying would cause.

It would still be trivial to copy data since you could open the 
network tab of the browser dev tools and copy the contents out of 
the web socket messages along with a few other workarounds.  I 
know this sort of thing is about making exfiltration harder, not 
impossible, but just something to be aware of.
